src/main/java/com/microsoftopentechnologies/windowsazurestorage/AzureBlobAction.java

```diff
@@ -154,13 +157,14 @@ public void doProcessDownloadRequest(
         response.sendError(Constants.HTTP_NOT_FOUND, "Azure artifact is not available");
     }
 
-    private String generateSASURL(StorageAccountInfo storageAccountInfo, String fileName) throws Exception {
+    private String generateReadSASURL(StorageAccountInfo storageAccountInfo, String fileName) throws Exception {
```

Comment:
can permission be a parameter to make this method general?

Reply:
There is a general method, only a wrapper here.